tensorflow_datasets/core/features/image_feature.py

```diff
@@ -345,3 +378,44 @@ def _get_and_validate_shape(shape, encoding_format):
         f'{acceptable_channels} (was {channels})'
     )
   return tuple(shape)
+
+
+def _get_and_validate_colormap(use_colormap, shape, encoding_format):
+  """Update the shape."""
+  acceptable_colormaps = [True, False]
+  if use_colormap not in acceptable_colormaps:
+    return ValueError(
+        'Invalid `use_colormap` choice.',
+        f'Available types: {acceptable_colormaps}. (was {use_colormap})'
+    )
+  if encoding_format and encoding_format != 'png':
+    raise ValueError(
+        f'Colormap is only available for PNG images. Not {encoding_format}'
+    )
+  if shape[-1] != 1 and use_colormap:
+    raise ValueError('Colormap is only available for gray-scale images')
+
+  return use_colormap
+
+
+def _apply_colormap(image: np.ndarray) -> np.ndarray:
```

Comment:
Could you add unittests for this function where you test the output on simple images ?

Reply:
Could you reference me to another tests? Would love a lead on how to do this.